app/assets/javascripts/components/amounttable.js

```diff
@@ -0,0 +1,313 @@
+ * @property {Any[]}    data      array of data to display (values are shown in order)
+ * @property {AmountTableColSpec[]} contents  An array of settings for each column (see above)
+ * @property {Number} [limit=Infinity]  The number of rows to show in collapsed state
+ * @property {function(data: Any): string} tooltip  A function specifying the tooltip content given a datarow
+ * @property {string}   tooltipID   A CSS selector of an existing tooltip element.
+ */
+
+/**
+ * @typedef {Object} AmountTableColSpec
+ * @property {function(data: Any): string} [html=null]
+ *           Function that generates the HTML content of the cell
+ *           (text is used if not supplied)
+ * @property {function(data: Any): string} [text=null]
+ *           Function that generates the HTML content of the cell
+ *           (html is preferred over text in when rendering a table, text is
+ *           preferred when exporting to csv)
+ * @property {object} [style=null]
+ *           object whose keys will be transformed to CSS properties of the
+ *           cells in the column
+ * @property {function(data: Any): number} [shade=false]
+ *           Function that calculates the amount the shader is filled for cells
+ *           in this column. Should be in [0,100]
+ */
+
+
+/**
+ * A table representation of data
+ */
+class AmountTable {
+    /**
+     * Constructor of an AmountTable
+     *
+     * Pro tip: use `style: {"background-color": "red"}` to change the shader
+     * fill color. Do not use `background` on a shaded cell.
+     *
+     * @param {AmountTableSettings} settings
+     */
+    constructor({el, title=null, header = null, data, contents=null, limit=Infinity, tooltip=null, tooltipID=null}) {
+        this.el = el;
+        this.title = title;
+        this.header = header;
+        this.data = data || [];
+        this.table = null;
+        this.limit = limit;
+        this.collapsed = true;
+        this.contents = contents;
+        this.makeTooltip = tooltip;
+        if (tooltip !== null) {
+            if (tooltipID === null) {
+                this.createTooltip();
+            } else {
+                this.tooltip = d3.select(tooltipID);
+            }
+        } else {
+            this.tooltip = null;
+        }
+    }
+
+    /**
+     * Create a tooltip element of class "tip" do display tooltips
+     */
+    createTooltip() {
+        this.tooltip = d3.select("body")
+            .append("div")
+            .attr("class", "tip")
+            .style("position", "absolute")
+            .style("z-index", "10")
+            .style("visibility", "hidden")
+            .style("background-color", "white")
+            .style("padding", "2px")
+            .style("border", "1px solid #dddddd")
+            .style("border-radius", "3px;");
+    }
+
+    /**
+     * Make the table visualization
+     */
+    draw() {
+        const [thead, tbody] = this.buildTable();
+        this.buildHeader(thead);
+        if (this.data.length != 0) {
```

Comment:
style: always use `!==` or `===`

_[Original comment](https://github.ugent.be/unipept/unipept/pull/625#discussion_r13713) by @bmesuere on Mon Feb 19 2018 at 13:37._